### PR TITLE
[Tests-Only ] webUI tests for non-admin user and guest user

### PR DESCRIPTION
**Pull request by swoichha**
## Description 
- webUI tests for non-admin user 
- webUI tests for guest user 

## Related Issue
- Fixes https://github.com/owncloud/ocis-accounts/issues/105

### CODE REVIEW

File: ui/tests/acceptance/stepDefinitions/accountsContext.js [hunk 1]
```diff
@@ -12,10 +12,18 @@ Then('user {string} should be displayed in the accounts list on the WebUI', asyn
...
 When('the user changes the role of user {string} to {string} using the WebUI', function (username, role) {
   return client.page.accountsPage().selectRole(username, role)
 })
 
 Then('the displayed role of user {string} should be {string} on the WebUI', function (username, role) {
   return client.page.accountsPage().checkUsersRole(username, role)
 })
+
+Then('the user should not be able to see the accounts list on the WebUI', function () {
+  return client.page.accountsPage().navigateToAccounts()
```
Comment (dpakach) [inline ui/tests/acceptance/stepDefinitions/accountsContext.js:28]: This function doesn't check that the user cannot see the accounts page.

File: ui/tests/acceptance/pageobjects/accountsPage.js [hunk 2]
```diff
@@ -40,6 +40,11 @@ module.exports = {
         util.format(this.elements.currentRole.selector, role)
 
       return this.useXpath().expect.element(roleSelector).to.be.visible
+    },
+
+    navigateToAccounts: async function () {
+      await this.waitForElementVisible('@loadingAccountsList')
+      return this.useXpath().expect.element('@accountsListTable').to.not.present
```
Comment (dpakach) [inline ui/tests/acceptance/pageobjects/accountsPage.js:47]: Fix the function name. also, we generally try not to do assertions on pageObjects.

Comment (swoichha) [reply ui/tests/acceptance/pageobjects/accountsPage.js]: Fixed

File: ui/tests/acceptance/stepDefinitions/accountsContext.js [hunk 3]
```diff
@@ -12,10 +12,22 @@ Then('user {string} should be displayed in the accounts list on the WebUI', asyn
...
 
 Then('the displayed role of user {string} should be {string} on the WebUI', function (username, role) {
   return client.page.accountsPage().checkUsersRole(username, role)
 })
+
+Then('the user should not be able to see the accounts list on the WebUI', async function () {
+  return client.page
+    .accountsPage()
+    .waitForElementVisible('@loadingAccountsList')
+    .expect.element('@accountsListTable')
```
Comment (kulmann) [inline ui/tests/acceptance/stepDefinitions/accountsContext.js:31]: This should involve some waiting, right? The loader is an intermediate state for an admin user as well, so we should have like a 5 seconds delay before we check if the accounts list was loaded.

Comment (swoichha) [reply ui/tests/acceptance/stepDefinitions/accountsContext.js]: Yeah, you are right. I have added 5 seconds timeout now before checking for accounts list.

File: ui/tests/acceptance/features/accounts.feature [hunk 4]
```diff
@@ -9,11 +9,26 @@ Feature: Accounts
...
+	Scenario: admin changes non-admin user's role to admin
 		Given user "Moss" has logged in using the webUI
 		When the user browses to the accounts page
 		Then user "einstein" should be displayed in the accounts list on the WebUI
 		When the user changes the role of user "einstein" to "Admin" using the WebUI
 		Then the displayed role of user "einstein" should be "Admin" on the WebUI
 		When the user reloads the current page of the webUI
 		Then the displayed role of user "einstein" should be "Admin" on the WebUI
+
+	Scenario: non-admin user should not be able to see accounts list
```
Comment (kulmann) [inline ui/tests/acceptance/features/accounts.feature:21]: Please rephrase `non-admin user` to `regular user`.

**Review by kulmann — CHANGES_REQUESTED** [review]
One minor wording proposal (see comment). Your PR needs a rebase to pass CI (because of this PR: https://github.com/owncloud/ocis-accounts/pull/110).

**Comment by kulmann** [comment]
Please note that we want to hide the accounts UI for non-admin users entirely in the future. Tracked in this ticket: https://github.com/owncloud/product/issues/167 We will have to change these tests again, then, but it's really good to have them like you implemented them now, thank you!

**Review by kulmann — APPROVED** [review]
nice! 🚀

**Review by individual-it — APPROVED** [review]

**Pull request merged**